cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -218,38 +221,32 @@ public Future<Void> rollingRestart(Function<Pod, RestartReasons> podNeedsRestart
         singleExecutor.submit(() -> {
             try {
                 LOGGER.debugCr(reconciliation, "Verifying cluster pods are up-to-date.");
-                List<NodeRef> pods = new ArrayList<>(nodes.size());
-
+                List<NodeRef> controllerPods = new ArrayList<>();
+                List<NodeRef> brokerPods = new ArrayList<>();
                 for (NodeRef node : nodes) {
                     // Order the nodes unready first otherwise repeated reconciliations might each restart a pod
                     // only for it not to become ready and thus drive the cluster to a worse state.
-
-                    // TODO: In KRaft mode, We currently roll only nodes with the broker role. This is because of Kafka
-                    //       limitations. Once managing controller nodes is supported with Kafka Admin API, this should be
-                    //       fixed. This is tracked in https://github.com/strimzi/strimzi-kafka-operator/issues/8593.
-                    if (node.broker()) {
-                        pods.add(podOperations.isReady(namespace, node.podName()) ? pods.size() : 0, node);
-                    } else if (node.controller())   {
-                        // We do not roll KRaft controllers yet, but we can throw a warning if it needs to be rolled
-                        Pod pod = podOperations.get(namespace, node.podName());
-
-                        if (pod != null)    {
-                            RestartReasons reasons = podNeedsRestart.apply(pod);
-
-                            if (reasons.shouldRestart())   {
-                                LOGGER.warnCr(reconciliation, "The controller node {} needs to be rolled because of {}. Strimzi currently does not support rolling of controller nodes. Please roll the node manually using 'kubectl delete pod {}'.", node.podName(), reasons.getReasons(), node.podName());
-                            }
-                        }
+                    // in KRaft mode roll unready controllers, then ready controllers, then unready brokers, then ready brokers
+                    boolean isReady = podOperations.isReady(namespace, node.podName());
+                    if (node.controller()) {
+                        controllerPods.add(isReady ? controllerPods.size() : 0, node);
+                    } else {
+                        brokerPods.add(isReady ? brokerPods.size() : 0, node);
                     }
                 }
+                LOGGER.debugCr(reconciliation, "Initial order for updating pods (rolling restart or dynamic update) is {},{}", controllerPods, brokerPods);
```

Comment:
I would clarify log with `..... is controller pods={}, broker pods={}`. Wdyt?

Reply:
Yeah I think that's a good idea